crates/network-scanner-net/src/async_io.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    num::NonZeroUsize,
+    os::windows::io::{AsRawSocket},
+    sync::Arc,
+};
+
+use polling::{Event, Events};
+use socket2::Socket;
+use tokio::sync::{Mutex};
+
+use crate::tokio_raw_socket::TokioRawSocket;
+
+#[derive(Debug)]
+pub struct AsyncIoRuntime {
+    poller: Arc<polling::Poller>,
```

Comment:
suggestion: Could be `poller: polling::Poller` if we can work with `Arc<AsyncIoRuntime>`.

Reply:
updated